Editor/Windows/AssetPaletteWindowFolderPanel.cs

```diff
@@ -164,6 +169,34 @@ private void InitializeFoldersTreeView()
                 foldersTreeView.CreateFolderRequestedEvent += HandleTreeViewCreateFolderRequestedEvent;
                 foldersTreeView.DroppedAssetsIntoFolderEvent += HandleTreeViewDroppedAssetsIntoFolderEvent;
             }
+
+            EnsureFoldersGuids();
+        }
+
+        private void EnsureFoldersGuids()
+        {
+            for (int startIndex = 0; startIndex < FoldersSerializedProperty.arraySize; startIndex++)
+            {
+                SerializedProperty folderProperty = FoldersSerializedProperty.GetArrayElementAtIndex(startIndex);
+                EnsureFolderIDRecursively(folderProperty);
+            }
+            
+            ApplyModifiedProperties();
+        }
+
+        private void EnsureFolderIDRecursively(SerializedProperty folderProperty)
+        {
+            if (!string.IsNullOrEmpty(folderProperty.FindPropertyRelative(GuidPropertyName).stringValue))
+                return;
+
+            folderProperty.FindPropertyRelative(GuidPropertyName).stringValue = Guid.NewGuid().ToString();
```

Comment:
The problem with using an integer that keeps ticking up is that when managing this integer like we can't use the array length for instance, because of children's folders. I didn't find a proper way of doing it, and also could be even harder to maintaining this between collections, to ensure unique IDs per folder. Yeah guids are less efficient than simple ints, but are more reliable in that sense.
We could reuse the LongGuid I created on SOC, that's like the best of both worlds.